### PR TITLE
[NativeAOT/ARM] Emit correct unwind codes for VPOP

### PR DESCRIPTION
**Pull request by filipnavara**
Tested on JIT tests

Unwind codes:
```
0x1c70 <MathCeilingSingle_ro_MathCeilingSingleTest_Program__TestEntryPoint>: @0x428
  Compact model index: 1
  0x01      vsp = vsp + 8
  0xc9 0x81 pop {D8-D9}
  0x84 0x8f pop {r4, r5, r6, r7, r11, r14}
  0xb0      finish

0x37370 <System.Globalization.CalendricalCalculationsHelper__EquationOfTime>: @0x9f54
  Compact model index: 1
  0x09      vsp = vsp + 40
  0xc9 0x86 pop {D8-D14}
  0x84 0x80 pop {r11, r14}
  0xb0      finish
```

Corresponding prologs:
```
00001c70 <_fram0MathCeilingSingle_ro_MathCeilingSingleTest_Program__TestEntryPoint>:
    1c70: 2d e9 f0 48   push.w  {r4, r5, r6, r7, r11, lr}
    1c74: 2d ed 04 8b   vpush   {d8, d9}
    1c78: 82 b0         sub     sp, #8

00037370 <_fram0System.Globalization.CalendricalCalculationsHelper__EquationOfTime>:
   37370: 2d e9 00 48   push.w  {r11, lr}
   37374: 2d ed 0e 8b   vpush   {d8, d9, d10, d11, d12, d13, d14}
   37378: 8a b0         sub     sp, #40
```

### CODE REVIEW

File: src/coreclr/jit/unwind.cpp [hunk 1]
```diff
@@ -165,7 +165,12 @@ void Compiler::unwindPushPopCFI(regNumber reg)
 #endif
         ;
 
+#if defined(TARGET_ARM)
+    createCfiCode(func, cbProlog, CFI_ADJUST_CFA_OFFSET, DWARF_REG_ILLEGAL,
+                  reg >= REG_FP_FIRST ? 2 * REGSIZE_BYTES : REGSIZE_BYTES);
+#else
     createCfiCode(func, cbProlog, CFI_ADJUST_CFA_OFFSET, DWARF_REG_ILLEGAL, REGSIZE_BYTES);
```
Comment (jkotas) [inline src/coreclr/jit/unwind.cpp:172]: Are we assuming that we are never going to see SIMD or floating-point registers here for other platforms? Assert may be nice.

**Review by jkotas — APPROVED** [review]
Thank you!

File: src/coreclr/jit/unwind.cpp (hunk 1 above)
Comment (filipnavara) [reply src/coreclr/jit/unwind.cpp:172]: That assumption is currently true. This is only called on ARM and in one branch of x64 with integer registers. I agree that assert would be nice...

Comment (jkotas) [inline src/coreclr/jit/unwind.cpp:172]:
```suggestion
    assert(reg < REG_FP_FIRST);
    createCfiCode(func, cbProlog, CFI_ADJUST_CFA_OFFSET, DWARF_REG_ILLEGAL, REGSIZE_BYTES);
```

Comment (filipnavara) [reply src/coreclr/jit/unwind.cpp:173]: Thanks!

**Pull request merged**